### ISSUE TITLE
When will the `modules.py` be completed?

### ISSUE THREAD

**Issue by howardyclo** [makarandtapaswi/BallClustering_ICCV2019#5]
Can you release the training code and loss functions? Otherwise we cannot reproduce the results from paper. Thanks.
